pkg/share/manila/shareoptions/backend.go

```diff
@@ -19,6 +19,6 @@ package shareoptions
 // BackendOptions contains backend-specific options
 type BackendOptions struct {
 	CSICEPHFSdriver string `name:"csi-driver" backend:"csi-cephfs" protocol:"CEPHFS"`
-
+	NFSShareClient  string `name:"nfs-share-client" backend:"nfs" protocol:"nfs"`
```

Comment:
Two points to consider here:
* those attribute tags are case-sensitive and since the documentation says `NFS`, let's make the protocol tag for `NFSShareClient` upper-case as well. You make a good point here though, it should be case-insensitive (otherwise the parser will get confused), I'll post a PR.
* you could add an `optional` tag here ([see this for example](https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/share/manila/shareoptions/common.go#L21)), making the whole tag field look like this `name:"nfs-share-client" backend:"nfs" protocol:"NFS" value:"default=0.0.0.0"`. This will simplify the code in the `NFS.GrantAccess` method.

Reply:
Good idea with the default setting. I will have to re-test the case for the protocol -- I think I originally had it uppercase (basically I copied the CSICEPHFSdriver string) but it didn't work.

Reply:
Most likely. All the other attribute values are lower case so it's pretty possible I just made a mistake. I tested with the requested changes and everything seems to be working fine now.

Reply:
Perfect! Yes the current behavior is really confusing and needs to be changed.